### PR TITLE
Implement more QEC experiments 

### PR DESCRIPTION
**Pull request by nelimee**
This PR introduces more QEC experiments in the `notebooks/` folder in order to test in real conditions the `tqec` package and validate it on more involved examples.

The final goal of this PR is to obtain a scalable experiment represented by the "corner":
1. Initialise a logical qubit
2. Extend it vertically and horizontally to obtain the "corner" shape
3. Move the logical qubit by performing the needed measurements
4. Measure the observable on the moved logical qubit

### CODE REVIEW

**Comment by nelimee** [comment]
Marking this PR as ready for review. More QEC experiments will be implemented to fix #50, #51, #52 and #53, but I think it is better to have one PR for each of the issues.

**Comment by smburdick** [comment]
Are you looking for feedback on the quality of the experiments themselves?

**Comment by afowler** [comment]
I suggested these experiments :-)

**Comment by nelimee** [comment]
> I suggested these experiments :-)

I added a note about that, sorry for not including it in the first place.

**Comment by nelimee** [comment]
I need to check that the auto-merging did not break anything, I will post a new comment when this is ready for merging.

**Comment by nelimee** [comment]
The main notebook of this PR, [notebooks/logical_qubit_extended_memory_experiment.ipynb](https://github.com/QCHackers/tqec/blob/1e19a25265cd4fd0e76984bc36d2e5829002d6bb/notebooks/logical_qubit_extended_memory_experiment.ipynb), is still running smoothly without any noticeable change.
I would be in favor of merging this PR, as previously noted. 
Just waiting for approval to merge.

**Comment by nelimee** [comment]
Ready to merge, waiting for an accept from someone to be able to merge.

**Comment by nelimee** [comment]
Fix #37, still waiting for someone to approve to let me merge.

**Comment by nelimee** [comment]
@inmzhang would you mind checking if this PR can be merged?

**Comment by inmzhang** [comment]
Sure, I will come back and review this in half an hour.

File: notebooks/logical_qubit_extended_memory_experiment.ipynb
Comment (inmzhang) [inline notebooks/logical_qubit_extended_memory_experiment.ipynb:1]: <p>The code generating the logical memory circuits looks good to me. Some ideas on the simulation part to gain more confidence on the correctness of implementation:</p><p> </p><ol><li>We can simulate for both of the X/Z logical basis to verify that the extended logical basis outperforms the regular one.</li><li>We can simulate a equal-distance square logical qubit memory experiment for comparison. If the implementation is correct, the extended logical basis should outperform that in the square case. And the regular logical basis should perform worse than that in the square case(there are more error paths for it to fail).</li></ol>
<br/>

 _Reply via <a href='https://app.reviewnb.com/QCHackers/tqec/pull/44/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

**Review by inmzhang — APPROVED** [review]

File: notebooks/logical_qubit_extended_memory_experiment.ipynb
Comment (nelimee) [reply notebooks/logical_qubit_extended_memory_experiment.ipynb:1]: I will implement these validation in a follow-up PR, thanks for the remarks!

**Pull request merged**
